app/Filament/Resources/ServerResource/Pages/CreateServer.php

```diff
@@ -247,7 +256,15 @@ public function form(Form $form): Form
                                                 $range = $start <= $end ? range($start, $end) : range($end, $start);
                                                 foreach ($range as $i) {
                                                     if ($i > 1024 && $i <= 65535) {
-                                                        $ports->push($i);
+                                                        if (Allocation::query()->where('ip', $get('allocation_ip'))->where('port', $portEntry)->exists()) {
```

Comment:
This needs to happen after the foreach loop because of the [N+1 problem](https://stackoverflow.com/questions/97197/what-is-the-n1-selects-problem-in-orm-object-relational-mapping)!